### ISSUE TITLE
Cabal should include build targets in `--list-options`

### ISSUE THREAD

**Issue by hvr** [haskell/cabal#1382]
This would simplify implementing [`(bash)Programmable Completion`](http://www.gnu.org/software/bash/manual/html_node/Programmable-Completion.html) for commands taking a build target, such as `cabal build`, `cabal test` or `cabal run`, as the current state would require to reinvent `cabal`'s strategy for locating the `.cabal` file and parsing the cabal file for potential targets (while not being able to detect currently disabled targets)

So basically this just requires to include the targets where applicable in the `--list-options` output.

**Comment by hvr** [comment]
See also #4070
